### DISCUSSION TITLE
Marksman compatibility

### DISCUSSION THREAD

**Discussion by codingluke** [facebook/docusaurus#9193] [Q&A] [answered]
Dear Docusaurus Team.
I use docusaurus together with [marksman](https://github.com/artempyanykh/marksman), a lsp for markdown documents.

I just figured out that marksman and docusaurus have different ways to generate anchor-slugs for the headings when adding `:emoji:` to the heading.

Given a title `# :smiley: This is a friendly header`, docusaurus sripes out the whole `:smiley:` part. The anchor-slug will be `#-this-is-a-friendly-header`. On the other hand marksman is only removing the special chars `:` and ends up with `#smiley-this-is-a-friendly-header`.

In my opinion the marksman parser is more intuitive. Also it is not starting a slug with an `-`. I also [opened a ticket](https://github.com/artempyanykh/marksman/issues/244) at their side. I think both projects could profit in working together! checkout [their vim integration demo <3](https://github.com/artempyanykh/marksman#demos-and-tutorials)

Best
Codingluke

**Comment by slorber** [comment, marked as answer]
Thanks @codingluke , this is indeed a bug, but I'm not sure how we can solve it easily.

Closing this discussion in favor of the bug issue I opened here: https://github.com/facebook/docusaurus/issues/9194
